### ISSUE TITLE
Shapefile not exporting (AGAIN!)

### ISSUE THREAD

**Issue by lin-d-hop** [DigitalCommons/land-explorer-front-end#191]
**Describe the bug**
Shapefile doesn't download.

Reported by a user.

**To Reproduce**
Share -> Export data -> Shapefile -> nothing!

**Expected behaviour**
I want a shapefile in my downloads and a UI notification.

**Screenshots/GIFs**
https://user-images.githubusercontent.com/8669848/225616941-6363f31b-5e6e-46a0-9982-6812c7798445.mp4

**Context (please complete the following information):**
Chrome on production

**Additional context**
Please add unit tests when you fix this.
It's another thing that breaks seemingly every release.

**Comment by lin-d-hop** [comment]
Hi @rogup @King-Mob 
Can we get this into the next release? This trumps other issues so if something needs to be bumped off that's okay.

**Comment by rogup** [comment]
I've repro'ed in production and the shapefile download occasionally works, but mostly fails with a 503 error. I can't seem to repro on staging.

From the production backend pm2 logs, when trying to download a map with just 1 marker:

**WORKING CASE**
```
Executing (default): SELECT `id`, `map_id`, `user_id`, `access`, `viewed`, `created_date` FROM `user_map` AS `UserMap` WHERE `UserMap`.`user_id` = 1382 AND `User
Map`.`map_id` = '337' LIMIT 1;
Executing (default): SELECT `id`, `name`, `data`, `deleted`, `created_date`, `last_modified`, `is_snapshot` FROM `map` AS `Map` WHERE `Map`.`id` = '337';
Executing (default): SELECT `idmap_memberships`, `map_id`, `item_type_id`, `item_id` FROM `map_memberships` AS `MapMembership` WHERE `MapMembership`.`map_id` = '
337' AND `MapMembership`.`item_type_id` = 0;
Executing (default): SELECT `idmarkers`, `name`, `description`, `data_group_id`, `location`, `uuid` FROM `markers` AS `Marker` WHERE `Marker`.`idmarkers` = 2644;
Executing (default): SELECT `idmap_memberships`, `map_id`, `item_type_id`, `item_id` FROM `map_memberships` AS `MapMembership` WHERE `MapMembership`.`map_id` = '
337' AND `MapMembership`.`item_type_id` = 1;
Executing (default): SELECT `idmap_memberships`, `map_id`, `item_type_id`, `item_id` FROM `map_memberships` AS `MapMembership` WHERE `MapMembership`.`map_id` = '
337' AND `MapMembership`.`item_type_id` = 2;
127.0.0.1: GET /api/user/map/download/337 --> 200
localhost
Listening on 0.0.0.0:4000
```
The download API route succeeds but the server restarts immediately after...

**FAILED CASE**
```
Executing (default): SELECT `id`, `map_id`, `user_id`, `access`, `viewed`, `created_date` FROM `user_map` AS `UserMap` WHERE `UserMap`.`user_id` = 1382 AND `UserMap`.`map_id` = '337' LIMIT 1;
Executing (default): SELECT `id`, `name`, `data`, `deleted`, `created_date`, `last_modified`, `is_snapshot` FROM `map` AS `Map` WHERE `Map`.`id` = '337';
Executing (default): SELECT `idmap_memberships`, `map_id`, `item_type_id`, `item_id` FROM `map_memberships` AS `MapMembership` WHERE `MapMembership`.`map_id` = '337' AND `MapMembership`.`item_type_id` = 0;
Executing (default): SELECT `idmarkers`, `name`, `description`, `data_group_id`, `location`, `uuid` FROM `markers` AS `Marker` WHERE `Marker`.`idmarkers` = 2643;
Executing (default): SELECT `idmap_memberships`, `map_id`, `item_type_id`, `item_id` FROM `map_memberships` AS `MapMembership` WHERE `MapMembership`.`map_id` = '337' AND `MapMembership`.`item_type_id` = 1;
Executing (default): SELECT `idmap_memberships`, `map_id`, `item_type_id`, `item_id` FROM `map_memberships` AS `MapMembership` WHERE `MapMembership`.`map_id` = '337' AND `MapMembership`.`item_type_id` = 2;
localhost
Listening on 0.0.0.0:4000
```

The server seems to restart during the download API route, somewhere during/after the `getMapPolygonsAndLines` method.

**Comment by rogup** [comment]
@King-Mob This is now fixed (without any code changes).
Unit tests wouldn't have caught the bug, and it's unlikely that any E2E tests would either, since it was a problem in the pm2 command.

But maybe it's still a good idea to do @lin-d-hop 's suggesting of adding unit tests for the download route, to help us avoid future bugs in this area, since it will be embarassing if it breaks again! Will leave this for you tomorrow, as we discussed on Discord :)

**Comment by lin-d-hop** [comment]
@rogup Do we need to do something either in deployment or a scheduled job to restarting pm2 with --watch?

**Comment by rogup** [comment]
@lin-d-hop
pm2 was watching changes in the files and restarting the server when it saw the temporary shapefiles that are created. I fixed the npm 'serve' command to stop pm2 watching files a while ago https://github.com/DigitalCommons/land-explorer-back-end/commit/862291076bea00887e1783955658e6b8e0e0c44c

However, this didn't take effect because when we deploy releases, we just restart pm2 with the same configuration rather than using the 'serve' command.

So, to fix this, I killed pm2 and started it afresh with the new configuration that doesn't watch files.

**TL;DR We don't need to do anything else.**

**Comment by lin-d-hop** [comment]
Agreed. But not a release blocker. UTs can have a new issue.
